lib/src/sdk/metrics/meter_provider.dart

```
@@ -0,0 +1,34 @@
+import 'package:opentelemetry/api.dart' as api;
+import 'package:opentelemetry/sdk.dart' as sdk;
+import 'package:logging/logging.dart';
+
+class MeterProvider implements api.MeterProvider {
+  final _meters = <String, api.Meter>{};
+  final _logger = Logger('opentelemetry.sdk.metrics.meterprovider');
+
+  @override
+  sdk.Meter get(String instrumentationScopeName,
+      {String instrumentationVersion = '',
+      String schemaUrl = '',
+      Map<String, String> attributes = const {}}) {
+    if (instrumentationScopeName == null || instrumentationScopeName == '') {
```

I think we should throw an error here, null or empty values aren't allowed: https://opentelemetry.io/docs/reference/specification/metrics/api/#instrument-naming-rule

---

I originally wrote this code to throw an ArgumentError but then read [this](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#get-a-meter)

> In case an invalid name (null or empty string) is specified, a working Meter implementation MUST be returned as a fallback rather than returning null or throwing an exception, its name property SHOULD keep the original invalid value, and a message reporting that the specified value is invalid SHOULD be logged.

I've adjusted the code to instead produce an appropriate log, complete with the stacktrace so that one can track down the offending code.